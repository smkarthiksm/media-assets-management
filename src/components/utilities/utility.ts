import { AudioFile } from '../../stubbs/files';

export const setJwtToken = (token: string) => {
  localStorage.setItem('jwt', token);
};

export const getJwtToken = () => {
  return localStorage.getItem('jwt');
};

export const removeJwtToken = () => {
  return localStorage.removeItem('jwt');
};

export const sampleJwtToken =
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IlNpbW9uIE5peG9uIiwiaWF0IjoxNTE2MjM5MDIyfQ.uAtBWmANdDtnj-heMCuWroQYzW0L-GUajjEeXYkvE38';

export const isAuthenticated = () => !!getJwtToken();

export const getTitleFromFile = (file: File) => file.name.split('.')[0];

export const getFileTypeFromFile = (file: File) =>
  file.type.split('/')[0] as 'audio' | 'video';

const getDisplayValue = (value: number) => (value < 10 ? `0${value}` : value);

export const getDurationInFormat = (duration: number) => {
  const minutes = Math.floor(duration / 60);
  const seconds = Math.ceil(duration - minutes * 60);

  const minutesToDisplay = getDisplayValue(minutes);
  const secondsToDisplay = getDisplayValue(seconds);

  return `${minutesToDisplay}:${secondsToDisplay}`;
};

export const areFilesInValid = (files: Partial<AudioFile>[]) =>
  !files.length ||
  files.some((file) => !file.album || !file.artist || !file.title);

export const transformFiles = (files: File[]) => {
  return files.map((file) => ({
    title: getTitleFromFile(file),
    album: '',
    artist: '',
    duration: '',
    fileType: getFileTypeFromFile(file),
  }));
};
