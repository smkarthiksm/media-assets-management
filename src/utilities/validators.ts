export const textValidation = (input: string) => {
  return /^[A-Za-z]+$/.test(input);
};

export const emailValidation = (input: string) => {
  return /\S+@\S+\.\S+/.test(input);
};
