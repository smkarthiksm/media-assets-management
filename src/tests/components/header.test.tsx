import '@testing-library/jest-dom';
import { renderWithProviders } from '../test-utility';
import HeaderComponent from '../../components/header/header';
import { fireEvent, screen } from '@testing-library/react';
import { getJwtToken, setJwtToken } from '../../utilities/utility';

describe('HeaderComponent', () => {
  test('should render', () => {
    const { container } = renderWithProviders(<HeaderComponent />);
    expect(container).toMatchSnapshot();
  });

  test('should show menu options when profile icon is clicked', () => {
    renderWithProviders(<HeaderComponent />);

    const profileIconButton = screen.getByRole('button');
    fireEvent.click(profileIconButton);
    const menuItem = screen.getByRole('menuitem', { name: 'Logout' });
    expect(menuItem).toBeInTheDocument();
  });

  test('should logout when option is clicked', () => {
    setJwtToken('test');
    renderWithProviders(<HeaderComponent />);

    const profileIconButton = screen.getByRole('button');
    fireEvent.click(profileIconButton);
    const menuItem = screen.getByRole('menuitem', { name: 'Logout' });

    Object.defineProperty(window, 'location', {
      writable: true,
      value: { reload: jest.fn() },
    });

    expect(getJwtToken()).toEqual('test');
    fireEvent.click(menuItem);
    expect(getJwtToken()).toBeNull();
  });
});
