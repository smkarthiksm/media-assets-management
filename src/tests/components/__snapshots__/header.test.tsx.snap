// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should render 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular d-flex justify-content-between css-hyum1k-MuiToolbar-root"
    >
      <div
        class="logo-container"
      >
        <span
          class="logo"
        />
        <h6
          class="d-inline mx-1"
        >
          Media Assets Management
        </h6>
      </div>
      <div
        class="d-flex align-items-center justify-content-end link-container"
      >
        <div>
          <a
            class="text-white m-4"
            href="/"
          >
            All
          </a>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium profile-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              alt="Simon Nixon"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="profile-image.png"
            />
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
